lib/dataset/query.js

```diff
@@ -338,6 +338,50 @@ define({
             return this.mergeOptions({distinct: args});
         },
 
+        /*   {
+         *     division: db.from('division').one();
+         *   }
+         *
+         */
+        eager: function(includeDatasets, model) {
+            this._eager = includeDatasets;
+            this.rowCb = (topLevelResults) => {
+                console.log(includeDatasets);
+                var eagerResults = {},
+                    modelResults = null;
+
+                if (model) {
+                    return model.load(topLevelResults).chain(modelResults => {
+                        var whens = Object.keys(includeDatasets).map(key => {
+                            return modelResults[key].chain(result => {
+                                eagerResults[key] = result.toObject();
+                                return result;
+                            });
+                        });
+
+                        return comb.when(whens).chain(()=>{
+                            return Object.assign(modelResults.toObject(), eagerResults);
```

Comment:
We should go ahead and use `comb.when` on the toObject incase a user overrides it in implementation